### PR TITLE
refactor: split some actions off setSettings

### LINKED ISSUE #2753: New action:setSetting
Create a new action that can atomically update a setting. The existing setSettings action has a large interface.

To implementer:

Investigate adjusting the setSettings interface to something more ergonomic.
Introduce new setSetting action that can update a single setting field.

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This splits the `setSetting` action apart into a number of smaller actions for clarity.

## Task Item

closes #2753

### CODE REVIEW

**Comment by beyackle** [comment]
Don't merge this yet; I need to fix the e2e tests a bit more.

File: Composer/packages/client/src/pages/publish/index.tsx [hunk 1]
```diff
@@ -339,8 +339,8 @@ const Publish: React.FC<PublishPageProps> = (props) => {
 
       if (result) {
         if (settings.publishTargets && settings.publishTargets.length > index) {
-          const target = settings.publishTargets.slice(0, index).concat(settings.publishTargets.slice(index + 1));
-          await actions.setSettings(projectId, { ...settings, publishTargets: target });
+          const targets = settings.publishTargets.slice(0, index).concat(settings.publishTargets.slice(index + 1));
+          await actions.setPublishTargets(targets);
```
Comment (lei9444) [inline Composer/packages/client/src/pages/publish/index.tsx:343]: Hi @beyackle I found the action use setPublishTarget, use setPublishTarget[s] will throw error. And the arguments is ( _, publishTarget).

Comment (lei9444) [reply Composer/packages/client/src/pages/publish/index.tsx:343]: I fixed these in #3550

Comment (beyackle) [reply Composer/packages/client/src/pages/publish/index.tsx:343]: Thanks for taking care of that.

**Pull request merged**